http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/SocketChannelResolver.java

```diff
@@ -41,6 +43,33 @@
     private SocketChannelResolver() {
     }
 
+    /**
+     * Attempts to determine the {@link ChannelFactory} class that corresponds to the given
+     * event loop group.
+     *
+     * @param eventLoopGroup the event loop group to determine the {@link ChannelFactory} for
+     * @return A {@link ChannelFactory} instance for the given event loop group.
+     */
+    public static ChannelFactory<? extends Channel> resolveSocketChannelFactoryClass(EventLoopGroup eventLoopGroup) {
```

Comment:
Do we still need this?

Reply:
Good catch. Updated `SharedSdkEventLoopGroup` to store `SdkEventLoopGroup` so that we can get default channel factory associated with the default event loop group.